python/lsst/analysis/tools/atools/genericBuild.py

```diff
@@ -0,0 +1,369 @@
+            self._add_flux(name=key, config=config)
+
+
+class MagnitudeXTool(MagnitudeTool):
+    """A Tool metrics/plots with a magnitude as the dependent variable."""
+
+    mag_x = Field[str](default="", doc="Flux (magnitude) field to bin metrics or plot on x-axis")
+
+    @property
+    def config_mag_x(self):
+        # This is a logic error: it shouldn't be called before finalize
+        assert self.mag_x in self.fluxes
+        return self.fluxes[self.mag_x]
+
+    def finalize(self):
+        super().finalize()
+        self._set_flux_default("mag_x")
+        key_mag = f"mag_{self.mag_x}"
+        subsets = (("xAll", "allSelector"), ("xGalaxies", "galaxySelector"), ("xStars", "starSelector"))
+        for name, key in subsets:
+            self._set_action(
+                self.process.filterActions,
+                name,
+                DownselectVector,
+                vectorKey=key_mag,
+                selector=VectorSelector(vectorKey=key),
+            )
+
+
+class SizeConfig(Config):
+    """Configuration for size vector(s) to be loaded and potentially plotted"""
+
+    has_moments = Field[bool](doc="Whether this size measure is stored as 2D moments", default=True)
+    key_size = Field[str](
+        doc="Size column(s) to compute/plot, including moment suffix as {suffix}",
+    )
+    log10_size = Field[bool](
+        default=True,
+        doc="Whether to compute/plot)log10 of the sizes",
+    )
+    name_size = Field[str](
+        default="size",
+        doc="Name of the size (e.g. for axis labels)",
+    )
+    scale_size = Field[float](
+        default=0.2,
+        doc="Factor to scale sizes (multiply) by",
+    )
+    unit_size = Field[str](
+        default="arcsec",
+        doc="Unit for sizes",
+    )
+
+    def modify_action(self, action: VectorAction) -> VectorAction:
+        if self.log10_size:
+            action = Log10Vector(actionA=action)
+        return action
+
+
+class SizeDefaultConfig(Config):
+    bulge = ConfigField[SizeConfig](doc="Bulge model size config")
+    disk = ConfigField[SizeConfig](doc="Disk model size config")
+    moments = ConfigField[SizeConfig](doc="Second moments size config")
+    shape_slot = ConfigField[SizeConfig](doc="Shape slot size config")
+
+
+class SizeTool(AnalysisTool):
+    """Compute various object size definitions in linear or log space."""
+
+    attr_prefix = Field[str](doc="Prefix to prepend to size names as attrs", default="size_", optional=False)
+    sizes_default = SizeDefaultConfig(
+        bulge=SizeConfig(key_size="{band}_bdReB", name_size="CModel Bulge R_{eff}", has_moments=False),
+        disk=SizeConfig(key_size="{band}_bdReD", name_size="CModel Disk R_{eff}", has_moments=False),
+        moments=SizeConfig(key_size="{band}_i{suffix}", name_size="Second moment radius"),
+        shape_slot=SizeConfig(key_size="shape_{suffix}", name_size="Shape slot radius"),
+    )
+    size_type = ChoiceField[str](
+        doc="The type of size to calculate",
```

Comment:
Here (and other config fields), missing period. However, if a few slip through, I'm not overly concerned.